Modules/QuestieInit.lua

```diff
@@ -0,0 +1,237 @@
+    didPlayerEnterWorld = true
+
+    -- Initialize the tracker
+    coroutine.yield()
+    QuestieTracker:Initialize()
+
+    local dateToday = date("%y-%m-%d")
+
+    if Questie.db.char.showAQWarEffortQuests and (Questie.db.char.aqWarningPrintDate == nil or Questie.db.char.aqWarningPrintDate < dateToday) then
+        Questie.db.char.aqWarningPrintDate = dateToday
+        C_Timer.After(2, function()
+            print("|cffff0000-----------------------------|r")
+            Questie:Print("|cffff0000The AQ War Effort quests are shown for you. If your server is done you can hide those quests in the General settings of Questie!|r");
+            print("|cffff0000-----------------------------|r")
+        end)
+    end
+
+    QuestieMenu:OnLogin()
+
+    if Questie.db.global.debugEnabled then
+        QuestieLoader:PopulateGlobals()
+    end
+
+    Questie.started = true
+
+end
+
+function QuestieInit:LoadDatabase(key)
+    if QuestieDB[key] then
+        coroutine.yield()
+        QuestieDB[key] = loadstring(QuestieDB[key]) -- load the table from string (returns a function)
+        coroutine.yield()
+        QuestieDB[key] = QuestieDB[key]() -- execute the function (returns the table)
+    else
+        Questie:Debug(DEBUG_DEVELOP, "Database is missing, this is likely do to era vs tbc: ", key)
+    end
+end
+
+function QuestieInit:LoadBaseDB()
+    -- reset townsfolk on all characters before compile
+    for _, char in pairs(QuestieConfig.char) do
+        char.townsfolk = nil
+    end
+
+    -- load NPC data
+    QuestieInit:LoadDatabase("npcData")
+    QuestieInit:LoadDatabase("npcDataTBC")
+
+    -- load object data
+    QuestieInit:LoadDatabase("objectData")
+    QuestieInit:LoadDatabase("objectDataTBC")
+
+    -- load quest data
+    QuestieInit:LoadDatabase("questData")
+    QuestieInit:LoadDatabase("questDataTBC")
+
+    -- load item data
+    QuestieInit:LoadDatabase("itemData")
+    QuestieInit:LoadDatabase("itemDataTBC")
+
+    if QuestieDB.questDataTBC then
+        -- we loaded the TBC db, alias the tables
+        QuestieDB.questData = QuestieDB.questDataTBC
+        QuestieDB.objectData = QuestieDB.objectDataTBC
+        QuestieDB.npcData = QuestieDB.npcDataTBC
+        QuestieDB.itemData = QuestieDB.itemDataTBC
+    end
+
+end
+
+-- called by the PLAYER_LOGIN event handler
+-- this function creates the coroutine that runs "InitAllModules"
+function QuestieInit:Init()
+    -- we do this here because its required for Questie:Error
+    Questie.db = LibStub("AceDB-3.0"):New("QuestieConfig", QuestieOptionsDefaults:Load(), true)
+
+    local initFrame = CreateFrame("Frame")
+    local routine = coroutine.create(QuestieInit.InitAllModules)
+    initFrame:SetScript("OnUpdate", function()
+        local status, error = coroutine.resume(routine)
+        if status then
+            if coroutine.status(routine) == "dead" then
+                initFrame:SetScript("OnUpdate", nil)
+            end
+        else
```

Comment:
What is the difference between `status` and the return value of `coroutine.status(routine)` ?